lib/instrumentation/modules/elasticsearch.js

```diff
@@ -40,11 +40,15 @@ module.exports = function (elasticsearch, agent, version, enabled) {
           }
           return original.apply(this, args)
         } else {
-          var p = original.apply(this, arguments)
-          p.then(function () {
-            span.end()
-          })
-          return p
+          return original.apply(this, arguments)
+            .then(function (originalP) {
+              span.end()
+              return originalP
+            })
+            .catch(function(err) {
```

Comment:
I would use the second argument of `then(...)` for the error handler rather than `catch(...)`. Using `catch(...)` creates an extra promise it doesn't need to create.